python/cudf/cudf/_lib/copying.pyx

```diff
@@ -163,6 +162,15 @@ def scatter(source, maps, target, bounds_check=True):
 
     result_cols = columns_from_table(&c_result_table)
 
+    for i, in_col in enumerate(target_cols):
+        if is_categorical_dtype(in_col.dtype):
+            result_cols[i] = column.build_categorical_column(
+                categories=in_col.cat().categories,
+                codes=result_cols[i],
+                mask=result_cols[i].mask,
+                ordered=in_col.cat().ordered
+            )
+
     del c_source_table
     del c_target_table
```

Comment:
Why are these needed?

Reply:
IIRC these are allocated via `table_from_columns` which uses the old `gdf_column` infrastructure and just does a raw malloc to allocate the table / column structures.

Reply:
yup